schemaPool.go

```diff
@@ -136,6 +137,31 @@ func (p *schemaPool) parseReferencesRecursive(document interface{}, ref gojsonre
 	return nil
 }
 
+// bsonDToMap converts a bson.D to a bson.M at all nesting levels
+// we cannot use the bson.D.Map() function because that is only one level
+func bsonDToMap(doc bson.D) map[string]interface{} {
+	newDoc := map[string]interface{}{}
+	for _, entry := range doc {
+		if subDoc, isSubDoc := entry.Value.(bson.D); isSubDoc {
```

Comment:
nit: using `ok` throughout this function in these type assertions -- the reason for `ok` can be implied by the code itself, the variable name seems unnecessary :D